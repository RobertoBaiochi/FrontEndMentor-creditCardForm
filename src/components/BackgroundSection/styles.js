import styled, { css } from 'styled-components';
import bgDesktop from '../../assets/images/bg-main-desktop.png';
import bgMobile from '../../assets/images/bg-main-mobile.png';

export const MainContainer = styled.main`
  ${({ theme }) => css`
    min-height: 100vh;
    margin-inline: auto;
    background-image: url(${bgMobile});
    background-position: top;
    background-repeat: no-repeat;
    background-size: 100% 35%;
    background-color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;

    @media ${theme.media.mediumQueries} {
      background-image: url(${bgDesktop});
      background-position: left;
      background-size: 35% 100%;
    }
  `};
`;
