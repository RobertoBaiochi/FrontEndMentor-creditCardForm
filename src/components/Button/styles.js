import styled, { css, keyframes } from 'styled-components';

const buttonAnimation = keyframes`
  0%, 100% {
    background-position: right;
  }
  50% {
    background-position: left;
  }
`;

export const ButtonStyle = styled.button`
 ${({ theme }) => css`
    width: 100%;
    margin-top: ${theme.spacings.small} ;
    padding: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.medium};
    font-family: ${theme.font.family.default};
    background: ${theme.colors.darkViolet};
    color: ${theme.colors.white};
    border: none;
    outline-color: transparent;
    border-radius: ${theme.spacings.xsmall};
    cursor: pointer;

    &:disabled {
      background: ${theme.colors.mediumViolet};

      @media (hover: hover) {
        &:hover {
          background: ${theme.colors.mediumViolet};
          cursor: not-allowed;
        }
      }
    }

    @media (hover: hover) {
      &:hover {
        background: linear-gradient(to right, ${theme.colors.darkViolet}, ${theme.colors.secondaryColor},
          ${theme.colors.secondaryColor}, ${theme.colors.darkViolet});
        background-size: 500%;
        animation: ${buttonAnimation} 6s ease-in-out infinite;
      }
    }
 `};
`;