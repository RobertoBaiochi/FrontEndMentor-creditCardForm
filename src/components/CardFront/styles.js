import styled, { css } from 'styled-components';
import frontCard from '../../assets/images/bg-card-front.png';

export const FrontCard = styled.div`
  ${({ theme }) => css`
    max-width: 447px;
    max-height: 245px;
    background-image: url(${frontCard});
    background-repeat: no-repeat;
    background-size: contain;
    padding: ${theme.spacings.medium};
    border: 2px solid gold;
  `};
`;

export const LogoCard = styled.img`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
  `};
`;

export const InputNumber = styled.input`
  ${() => css`
    
  `}
`;
