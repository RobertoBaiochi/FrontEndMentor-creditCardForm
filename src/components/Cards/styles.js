import styled, { css } from 'styled-components';
import cardFront from '../../assets/images/bg-card-front.png';
import cardBack from '../../assets/images/bg-card-back.png';


export const CardFrontContainer = styled.div`
  ${() => css`
    width: 29rem;
    height: 17rem;
    background-image: url(${cardFront});
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    position: absolute;
    bottom: 0;
    left: 0;
    z-index: 2;
    border-radius: 1rem;
    padding: 1.5rem;
  `};
`;

export const CardFrontInfo = styled.div`
  ${() => css`
    width: 100%;
    height: 100%;

    > img {
      width: 25%;
      display: block;
      margin-bottom: 4rem;
    }

    > h3 {
      text-align: center;
      font-size: 2rem;
      word-spacing: .3rem;
      letter-spacing: .12rem;
      color: white;
      margin-bottom: 1.5rem;

    }

    > div {
      display: flex;
      justify-content: space-between;

      > span {
        font-size: 1.3rem;
        color: white;
        text-transform: uppercase;
        letter-spacing: .1rem;
      }
    }


  `};
`;


export const CardBackContainer = styled.div`
  ${() => css`
    width: 29rem;
    height: 17rem;
    background-image: url(${cardBack});
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    position: absolute;
    top: 0;
    right: 0;
    border-radius: 1rem;
    padding: 1.5rem;
  `};
`;

export const CardBackInfo = styled.div`
  ${() => css`
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: flex-end;

    > span {
      padding-right: 1.5rem;
      padding-bottom: 0.4rem;
      color: white;

    }
  `};
`;
