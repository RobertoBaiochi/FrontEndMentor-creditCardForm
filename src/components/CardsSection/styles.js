import styled, { css } from 'styled-components';

export const ContainerCards = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.medium} ${theme.spacings.xsmall};
    border: 8px solid yellow;
    position: relative;
    display:grid;
    grid-template-rows: repeat(2, 1fr);
    align-content: center;
    margin: ${theme.spacings.medium} auto ;

    @media ${theme.media.mediumQueries} {
      width: 50%;
      gap: 2rem;
    }
  `};
`;
