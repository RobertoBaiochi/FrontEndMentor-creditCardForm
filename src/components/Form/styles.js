import styled, { css, keyframes } from 'styled-components';

const inputAnimation = keyframes`
  0%, 100% {
    background-position: right;
  }
  50% {
    background-position: left;
  }
`;

const inputInvalid = keyframes`
  25% {
    translate: 8px 0;
  }
  50% {
    translate: -16px 0;
  }
  75% {
    translate: 8px 0;
  }

`

export const FormSection = styled.form`
  ${({ theme }) => css`
    width: 100%;
    margin: ${theme.spacings.medium} 0;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;

    @media ${theme.media.mediumQueries} {
      padding: 2rem;
    }
  `};
`;

export const FormContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    position: relative;

    > label {
      display: block;
      text-transform: uppercase;
      padding: ${theme.spacings.xsmall} 0;
      font-size: ${theme.font.sizes.medium};
      letter-spacing: 0.15rem;

    }

    > input {
      display: block;
      width: 100%;
      padding: ${theme.spacings.xsmall};
      border: 3px solid ${theme.colors.lightViolet};
      border-radius: ${theme.spacings.xsmall};
      font-family: ${theme.font.family.default};
      font-size: 2rem;
      text-transform: capitalize;
      margin-bottom: 1rem;


      &:focus {
        padding: .5rem;
        border: double 3px transparent;
        border-radius: 6px;
        background-image: linear-gradient( white, white),
          linear-gradient(to right, ${theme.colors.secondaryColor}, ${theme.colors.primaryColor},
          ${theme.colors.primaryColor}, ${theme.colors.secondaryColor});
        background-origin: border-box;
        background-clip: padding-box, border-box;
        background-size: 500%;
        animation: ${inputAnimation} 2s ease-in-out infinite;
        outline: transparent;
      }

      &::placeholder {
        font-family: ${theme.font.family.default};
        font-size: 1.8rem;
        font-weight: bold;
        color: ${theme.colors.lightViolet};
      }

    }

    .erro {
        border: solid 3px red;
        animation: ${inputInvalid} 200ms ease-in-out;
      }

  `};
`;

export const ErrorMsg = styled.small`
  ${({ theme }) => css`
    display: block;
    font-size: 1.3rem;
    color: ${theme.colors.redErrors};

    position: absolute;
    left: 0;
    bottom: -.5rem;
  `};
`;

export const DateCvcContainer = styled.div`
  ${() => css`
    display: flex;
    gap: 1rem;
  `};
`;

export const DateContainer = styled(FormContainer)`
  ${({ theme }) => css`

    width: 50%;

    > input {
      display: inline;
      width: 43%;
      margin-right: ${theme.spacings.small};
    }

  `};
`;

export const CvcContainer = styled(FormContainer)`
  ${() => css`
    width: 50%;
  `};
`;

export const Button = styled.button`
 ${({ theme }) => css`
    width: 100%;
    margin-top: ${theme.spacings.small} ;
    padding: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.medium};
    font-family: ${theme.font.family.default};
    background: ${theme.colors.darkViolet};
    color: ${theme.colors.white};
    border: none;
    outline-color: transparent;
    border-radius: ${theme.spacings.xsmall};
    cursor: pointer;

    &:disabled {
      background: ${theme.colors.mediumViolet};

      @media (hover: hover) {
        &:hover {
          background: ${theme.colors.mediumViolet};
          cursor: not-allowed;
        }
      }
    }

    @media (hover: hover) {
      &:hover {
        background: linear-gradient(to right, ${theme.colors.darkViolet}, ${theme.colors.secondaryColor},
          ${theme.colors.secondaryColor}, ${theme.colors.darkViolet});
        background-size: 500%;
        animation: ${inputAnimation} 6s ease-in-out infinite;
      }
    }
 `};
`;
