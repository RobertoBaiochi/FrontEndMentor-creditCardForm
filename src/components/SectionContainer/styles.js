import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme }) => css`
    max-width: 110rem;
    width: 100%;
    margin-inline: auto;
    padding: ${theme.spacings.medium} ${theme.spacings.small};
    border: 4px solid red;
  `};
`;
