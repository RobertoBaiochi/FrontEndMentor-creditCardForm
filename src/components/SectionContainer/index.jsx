import P from 'prop-types';
import { Container } from './styles';

function SectionContainer({ children }) {
  return (
    <Container>
      {children}
    </Container>
  );
}

export default SectionContainer;

SectionContainer.propTypes = {
  children: P.node.isRequired,
};
