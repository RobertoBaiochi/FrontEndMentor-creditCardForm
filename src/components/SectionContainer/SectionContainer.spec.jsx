import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import SectionContainer from '.';

describe('<SectionContainer />', () => {
  it('should render', () => {
    renderTheme(
      <SectionContainer>
        <h1>Testando</h1>
      </SectionContainer>);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
