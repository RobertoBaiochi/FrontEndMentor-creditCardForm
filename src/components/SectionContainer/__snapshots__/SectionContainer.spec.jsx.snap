// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> should render a children 1`] = `
<section
  class="sc-bcXHqe cyKDts"
>
  <h1>
    Children
  </h1>
</section>
`;
